eb_hooks.py

```diff
@@ -744,6 +744,25 @@ def pre_configure_hook_LAMMPS_zen4(self, *args, **kwargs):
     else:
         raise EasyBuildError("LAMMPS-specific hook triggered for non-LAMMPS easyconfig?!")
 
+def pre_configure_hook_nvhpc_x86_64(self, *args, **kwargs):
+    """
+    pre-configure hook for nvhpc
+    - search and replace operations in the ec dict
+    """
+    if self.name == "NVHPC":
+        eprefix = get_eessi_envvar("EPREFIX")
+        new_opts = f'''installdir=%(installdir)s/Linux_x86_64/%(version)s
+            EPREFIX={eprefix}
+            sed -i "s@\(set LDSO=.*\);@\\1 --sysroot=$EPREFIX;@" $installdir/compilers/bin/localrc
+            echo "set DEFLIBDIR=$EPREFIX/usr/lib64;" >> $installdir/compilers/bin/localrc
+            echo "set DEFSTDOBJDIR=$EPREFIX/usr/lib64;" >> $installdir/compilers/bin/localrc'''
```

Comment:
Thinking about this a bit more, the logic here could be added directly to the NVHPC easyblock (and used conditionally based on whether the EB build option `--sysroot` is set).